auth/gcloud/aio/auth/session.py

```diff
@@ -79,6 +79,14 @@ async def delete(
     ) -> Response:
         pass
 
+    @abstractmethod
+    async def head(
+        self, url: str, headers: Optional[Mapping[str, str]],
+        timeout: float, params: Optional[Mapping[str, Union[int, str]]],
+        allow_redirects: bool,
+    ) -> Response:
+        pass
```

Comment:
Perhaps https://docs.python.org/3/library/exceptions.html#NotImplementedError?

Reply:
That could definitely work, although I would like to limit the sprawl of my changes and this is an extension of existing form. Since this is an abstract base class and this is an abstract method, you can't instantiate an object of this class so raising is redundant:

```python
>>> import abc
>>> class Base(abc.ABC):
...     def fn(self):
...         return 42
...     @abc.abstractmethod
...     def abstract(self):
...         pass
... 
>>> b = Base()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Can't instantiate abstract class Base with abstract method abstract
```